tools/clang-format/clang-format-diff.py

```diff
@@ -114,15 +132,17 @@ def main():
       sys.exit(p.returncode)
 
     if not args.i:
-      with open(filename) as f:
-        code = f.readlines()
+      tmpfile.seek(0)
```

Comment:
if you no longer open the file/keep the common path, then you have already recorded the content, you can pass it directly rather than reading from the disk

Reply:
clang-format (called at line 125) needs the file name anyway. I can also pass the content in the tuple.